pyweek.py

```diff
@@ -130,6 +127,113 @@ def download(challenge, directory):
         )
 
 
+@cli.command()
+@click.argument(
+    'file',
+    type=Path,
+)
+def verify(file: Path):
+    """
+    Determines if a given zip file is in the proper format.
+    """
+
+    errors = 0
+
+    if not file.exists():
+        error(f"File {file} does not exist.")
+        sys.exit()
+
+    if not file.suffix == '.zip':
+        error(f"File is not a zip file.", errors)
+        errors += 1
+
+    # Check that the file name follows the proper naming convention
+    pattern = re.compile(r'^[A-Za-z0-9-]+-[0-9]+\.[0-9]+(\.[0-9]+)?\.zip$')
+    match = pattern.match(file.name)
+    if not match:
+        error(f"""File does not follow the proper naming convention.
+    The file name should be in the format: {{Name-of-Entry}}-{{major.minor}}.zip
+    Example: "My-Game-1.0.zip" or "my-game-1.0.1.zip\"""", errors)
+        errors += 1
+
+    # Open the zip file
+    try:
+        zipped_file = zipfile.ZipFile(file)
+    except zipfile.BadZipFile:
+        error(f"File is not a valid zip file.", errors)
+        sys.exit(1)
+    except IsADirectoryError:
+        error(f"File is a directory.", errors)
+        sys.exit(1)
+
+    # Check that the zip file contains a single top-level directory
+    # This directory should be named the same as the zip file
+    top_level_dirs = []
+    for name in zipped_file.namelist():
+        name = name.split('/')[0]
+        if name not in top_level_dirs:
+            top_level_dirs.append(name)
```

Comment:
This could be a set comprehension:


```suggestion
    top_level_dirs = {
        name.split('/')[0]
        for name in zipped_file.namelist()
    }
```